### PR TITLE
feat(vg_lite): add image clip corner support

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

Added support for rounded corner cropping of images without performance loss.

image

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/draw/lv_draw.h [hunk 1]
```diff
@@ -165,6 +165,9 @@ struct _lv_layer_t  {
      */
     lv_area_t _clip_area;
 
+    /** Clip radius of the layer. Used for rounded corners */
+    int32_t clip_radius;
```
Comment (kisvegabor) [inline src/draw/lv_draw.h]: As it can be allied on normal images too (not only layers), I think it should be part of `lv_draw_image_dsc_t`.

Comment (FASTSHIFT) [reply src/draw/lv_draw.h]: Done

File: src/widgets/image/lv_image.c [hunk 2]
```diff
@@ -749,6 +749,11 @@ static void draw_image(lv_event_t * e)
                         obj->coords.y1,
                         obj->coords.x1 + img->w - 1,
                         obj->coords.y1 + img->h - 1);
+
+            if(lv_obj_get_style_clip_corner(obj, LV_PART_MAIN)) {
```
Comment (kisvegabor) [inline src/widgets/image/lv_image.c]: I think it should be independent of `clip_corner`.

- If only `clip_radius` is only the image will have radius
- If `clip_corner` is set too, the the children will be clipped as well. It uses a lot of memory for large radiuses, so I would leave to the user to control it.

**Comment by kisvegabor** [comment]
Can you update some vector graphics tests to be sure it's working?

File: src/widgets/image/lv_image.c (hunk 2 above)
Comment (FASTSHIFT) [reply src/widgets/image/lv_image.c]: Do you mean to add a separate API to image: `lv_image_set/get_clip_radius`?

**Comment by FASTSHIFT** [comment]
> Can you update some vector graphics tests to be sure it's working?

I finally figured out how to add a new reference picture group~
VG-Lite test here: https://github.com/lvgl/lvgl/pull/6264

**Comment by FASTSHIFT** [comment]
@kisvegabor 

> Can you update some vector graphics tests to be sure it's working?

I finally figured out how to add an independent reference image group~
VG-Lite test here: https://github.com/lvgl/lvgl/pull/6264.

File: src/widgets/image/lv_image.c (hunk 2 above)
Comment (kisvegabor) [reply src/widgets/image/lv_image.c]: I'm sorry I missed this comment. I meant just removing the `if(lv_obj_get_style_clip_corner(obj, LV_PART_MAIN)) {` condition so the radius is always applied if set.

**Review by kisvegabor — COMMENTED** [review]
Sorry for the delay. It looks good, just one comment.

File: src/widgets/image/lv_image.c (hunk 2 above)
Comment (FASTSHIFT) [reply src/widgets/image/lv_image.c]: Done

**Comment by FASTSHIFT** [comment]
> Can you update some vector graphics tests to be sure it's working?

Done

**Review by XuNeo — APPROVED** [review]

**Review by kisvegabor — APPROVED** [review]

**Pull request merged**